### PR TITLE
Various Swagger doc fixes

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-3351](https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3351)

### Context

There are a number of bugs/discrepancies in the swagger documentation that we want to address.

### Changes proposed in this pull request

- Fix funded_place description/add missing funded_place
- Update change-funded-place response example
- Make UnprocessableEntityResponse schema generic
- Fix enrolled typo in swagger docs
- Restrict sorting swagger docs to endpoints that support it
- Add missing data attribute to declaration request
- Add request filters to v1/2 declarations CSV endpoint
- Set clawback_statement_id to nil in v3 void response
- Add voided state to ParticipantOutcome response schema
- Fix error on collapsing participant outcomes swagger accordion
- Fix completion_date format in swagger docs

### Guidance to review

Best reviewed by-commit.

### CODE REVIEW

File: spec/swagger_schemas/models/enrolment_csv.rb [hunk 1]
```diff
@@ -50,6 +50,12 @@
         type: :boolean,
         example: true,
       },
+      funded_place: {
+        description: "Indicates whether this NPQ participant is funded by DfE",
+        type: :boolean,
+        example: true,
+        required: true,
+      },
```
Comment (mooktakim) [inline spec/swagger_schemas/models/enrolment_csv.rb:58]: Can we move this down below `school_urn` to maintain the same ordering as the serializer.

Comment (ethax-ross) [reply spec/swagger_schemas/models/enrolment_csv.rb]: Updated 👍

**Review by leandroalemao — APPROVED** [review]
well done!! lgtm 👍  tks

File: spec/requests/api/docs/v3/applications_spec.rb [hunk 2]
```diff
@@ -71,19 +71,25 @@
...
+                    "#/components/schemas/ApplicationChangeFundedPlaceRequest" do
+      let(:application) { create(:application, :eligible_for_funded_place, lead_provider:, schedule:, cohort:, course:) }
+      let(:resource) { application }
+      let(:type) { "npq-application-change-funded-place" }
+      let(:attributes) { { funded_place: true } }
+      let(:invalid_attributes) { { funded_place: nil } }
+      let(:response_example) do
+        base_response_example.tap do |example|
+          example[:data][:attributes][:status] = "accepted"
+          example[:data][:attributes][:funded_place] = true
```
Comment (cwrw) [inline spec/requests/api/docs/v3/applications_spec.rb:91]: could we also this on the accepted endpoint and set the `funded_place` to true :)?

**Review by cwrw — APPROVED** [review]
minor comment otherwise looks great and thanks for the commit breakdown 💯

**Pull request merged**